sycl/source/detail/sampler_impl.cpp

```diff
@@ -24,15 +24,14 @@ sampler_impl::sampler_impl(coordinate_normalization_mode normalizationMode,
   verifyProps(MPropList);
 }
 
-sampler_impl::sampler_impl(cl_sampler clSampler,
-                           const ContextImplPtr &syclContext) {
-  const AdapterPtr &Adapter = syclContext->getAdapter();
+sampler_impl::sampler_impl(cl_sampler clSampler, context_impl &syclContext) {
+  const AdapterPtr &Adapter = syclContext.getAdapter();
   ur_sampler_handle_t Sampler{};
   Adapter->call<UrApiKind::urSamplerCreateWithNativeHandle>(
       reinterpret_cast<ur_native_handle_t>(clSampler),
-      syclContext->getHandleRef(), nullptr, &Sampler);
+      syclContext.getHandleRef(), nullptr, &Sampler);
 
-  MContextToSampler[syclContext] = Sampler;
+  MContextToSampler[syclContext.shared_from_this()] = Sampler;
```

Comment:
`std::unordered_map::operator[]` was performing a copy internally before, now it can just move (and rvalue-ref overload is C++11, so available to us), so no significant (if at all) overhead here.